File: Coding_Problems/Arrays,Strings,Linkedlists/plusOne.cpp
#include<iostream>
#include <bits/stdc++.h>
#include<vector>

using namespace std;


class Solution {
public:
    vector<int> plusOne(vector<int>& digits) {
         for(int i = digits.size()-1; i>=0; i--){
                    if(digits[i]<9){
                    digits[i]++;
                    return digits;
                    }
                    else
                            digits[i] = 0;
                    
            }
            digits.push_back(0);
            digits[0] = 1;
            return digits;
        
    }
};
int main(){
    Solution obj;
    vector<int> nums;
    nums.push_back(9);
    nums.push_back(9);
    nums.push_back(9);
    obj.plusOne(nums);
    for(int i=0;i<nums.size();i++){
        cout<<nums[i];
    }
    return 0;
}

File: Coding_Problems/Arrays,Strings,Linkedlists/plusOne.py
def plusOne(self, digits):
        
    length =len(digits) 
    num=0
    res=[]
    
    #[1,2,3] i=1 10^2 , i=2*10^1 , i =3*10^0
    for i in digits:
        num =num + i*pow(10,length-1) 
        length-=1  # 2 , 1, 0
    num+=1
    num=str(num)
    res =[int(x) for x in num]
    return res
        
